Firestore/core/src/model/server_timestamp_util.cc

```diff
@@ -57,7 +59,7 @@ google_firestore_v1_Value EncodeServerTimestamp(
   if (previous_value) {
     ++field;
     field->key = nanopb::MakeBytesArray(kPreviousValueKey);
-    field->value = DeepClone(*previous_value);
+    field->value = *DeepClone(*previous_value).release();
```

Comment:
To be clear, I meant having both (signatures might be slightly off):
```cpp
T* DeepCloneRaw(const T&); // Use in places like this, where otherwise you'd immediately call `release`
Message<T> DeepClone(const T&); // Use everywhere else
```

Feel free to postpone or push back.

Reply:
I prefer keeping it as is for now with just one function.